library/core/src/num/int_macros.rs

```diff
@@ -2477,18 +2477,18 @@ macro_rules! int_impl {
                 None
             } else {
                 let mut n = 0;
-                let mut r = self;
+                let mut r = 1;
 
                 // Optimization for 128 bit wide integers.
                 if Self::BITS == 128 {
```

Comment:
For the 128-bit sections in both functions, I was confused as to the reason for `Self::ilog2(base) + 1`. To my understanding, it's to ensure an accurate lower bound of the actual result for, say, `14.checked_ilog(15)`. A one line `//` comment would be helpful there. Also, `n` and `r` can be assigned to directly since they're initialized to the `+` and `*` identities, respectively, though I doubt it makes a difference with optimizations.

Reply:
0c80243e149e522f28eb3650f42d82a0c5226491 adds a comment explaining the correctness of the 128 bit optimization